### ISSUE TITLE
Password request for an unprotected key

### ISSUE THREAD

**Issue by beldmit** [openssl/openssl#16197]
The following commands
```
openssl genrsa -out key.pem 2048
openssl req -new -key key.pem -out test.csr
```
request a password for the unprotected key

It's a regression comparing OpenSSL 1.1.1

**Comment by beldmit** [comment]
The `openssl-req` command `-keyout` documentation denotes

```
This gives the filename to write any private key to that has been newly created or read from B<-key>.
```

In 1.1.1 it was
```
This gives the filename to write the newly created private key to.
```
Is that a desired change?

**Comment by DDvO** [comment]
Yes, this copying of the key to `-keyout` was intended, but not that the implicit way of providing the file name via the `default_keyfile` configuration options leads to an unexpected password prompt.

**Comment by schopin-pro** [comment]
I don't know if that's exactly the same issue, or even an actual one, but this works in 1.1.1l without password prompt, but not in 3.0.0:

```sh
openssl genrsa -out key.pem 2048
openssl req -new -key key.pem -keyout key.pem -out test.csr
```

This pattern can be found in the easy-rsa wrapper, and causes OpenVPN/easy-rsa#454

**Comment by mattcaswell** [comment]
@schopin-pro - this looks like the same issue as far as I can tell. It's already fixed in git.

**Comment by schopin-pro** [comment]
AFAICT the patch is even included in the 3.0.0 release. However, the issue is fixed for the OP's case, where they didn't provide `-keyout` but my issue is when `-keyout` is provided, and points to the same file as `-key`. It behaves as documented : in 1.1.1l since -key is provided, it doesn't do anything, but in 3.0.0 it actually tries to write out the key, which is when the passphrase prompt kicks in, as I understand it.

I don't think this should be fixed as it seems more like a misuse to me, but the '-keyout' change might be worth mentioning in the migration guide?
